src/components/FeaturePanel/Climbing/CragsInAreaFilter.tsx

```diff
@@ -0,0 +1,224 @@
+  const currentGradeSystem = userSettings['climbing.gradeSystem'] || 'uiaa';
+  const values = GRADE_TABLE[currentGradeSystem];
+  const uniqueValues = useMemo(() => [...new Set(values)], [values]);
+
+  return {
+    uniqueValues,
+    gradeInterval,
+    setGradeInterval,
+    minimumRoutesInInterval,
+    setMinimumRoutesInInterval,
+    setIsTouched,
+    isTouched,
+  };
+};
+
+export const filterCrag =
+  ({
+    gradeInterval,
+    currentGradeSystem,
+    uniqueValues,
+    minimumRoutesInInterval,
+  }) =>
+  (crag) => {
+    if (!gradeInterval) return true;
+
+    const numberOfFilteredRoutes = crag.memberFeatures.reduce((acc, route) => {
+      const difficulty = getDifficulty(route.tags);
+      if (!difficulty) return acc;
+      const convertedGrade = convertGrade(
+        difficulty.gradeSystem,
+        currentGradeSystem,
+        difficulty.grade,
+      );
+      if (
+        isInGradeInterval({
+          gradeMin: uniqueValues[gradeInterval[0]],
+          gradeMax: uniqueValues[gradeInterval[1]],
+          grade: convertedGrade,
+          currentGradeSystem,
+        })
+      )
+        return acc + 1;
+      return acc;
+    }, 0);
+    return numberOfFilteredRoutes > minimumRoutesInInterval;
+  };
+
+export const CragsInAreaFilter = ({
+  gradeInterval,
+  setGradeInterval,
+  minimumRoutesInInterval,
+  setMinimumRoutesInInterval,
+  uniqueValues,
+  setIsTouched,
+  isTouched,
+}) => {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+
+  const open = Boolean(anchorEl);
+  const { userSettings, setUserSetting } = useUserSettingsContext();
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleIfTouched = ({
+    gradeInterval,
+    minimumRoutesInInterval,
+  }: {
+    gradeInterval?: number[];
+    minimumRoutesInInterval?: number;
+  }) => {
+    if (
+      gradeInterval &&
+      (gradeInterval[0] !== 0 || gradeInterval[1] !== uniqueValues.length - 1)
+    )
+      setIsTouched(true);
+    else if (minimumRoutesInInterval && minimumRoutesInInterval !== 1)
+      setIsTouched(true);
+    else {
+      setIsTouched(false);
+    }
+  };
+
+  const handleChangeGradeFilter = (
+    _event: Event,
+    newValue: number | number[],
+  ) => {
+    if (Array.isArray(newValue)) {
+      handleIfTouched({ gradeInterval: newValue });
+      setGradeInterval(newValue);
+    }
+  };
+  const handleChangeMinimumRoutesInInterval = (
+    _event: Event,
+    newValue: number,
+  ) => {
+    handleIfTouched({ minimumRoutesInInterval: newValue });
+    setMinimumRoutesInInterval(newValue);
+  };
+
+  const currentGradeSystem = userSettings['climbing.gradeSystem'] || 'uiaa';
+
+  useEffect(() => {
+    if (gradeInterval === null) setGradeInterval([0, uniqueValues.length - 1]);
+  }, [uniqueValues, gradeInterval, currentGradeSystem, setGradeInterval]);
```

Comment:
I would probably bundle this useEffect in custom hook together with the L25 (`const [gradeInterval, setGradeInterval] = React.useState...`) and also L32-34 (the unique values from grade system), because it logicaly goes together. 

I was at first rather confused what should exactly go in the `gradeInterval`.